File: practica3/PYTHON/Suavizado/medianaSuavizado.py
import cv2
import numpy as np

filename = "../../data/images/salt-and-pepper.png"

img = cv2.imread(filename)
# Comprobamos se se leu a imaxe
if img is None:
    print("Non poiden ler a imaxe")

# Definimos o tamaño do kernel
kernelSize = 5

# Executamos o filtrado da mediana e o almacenamos nun array de numpy
medianBlurred = cv2.medianBlur(img,kernelSize)

cv2.imshow("Imaxe Orixinal", img)
cv2.waitKey(0)
cv2.imshow("Filtro da mediana : KernelSize = 5", medianBlurred)
cv2.waitKey(0)

File: codigos_clase/practica4/CODIGO/C++/Suavizado/medianaSuavizado.cpp
#include <iostream>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(){
string filename = "../../../data/images/gaussian-noise.png";
Mat image = imread(filename);

// Definimos o tamaño do kernel
int kernelSize = 5;

Mat medianBlurred;
// Realizamos o suavizado da mediana e almacenamos o resultado nun array de numpy
medianBlur(image,medianBlurred,kernelSize);

imshow("Imaxe Orixinal",image);
waitKey(0);
imshow("Filtro da mediana : KernelSize = 5",medianBlurred);
waitKey(0);
return 0;
}
